docs/source/developers/python.rst

```diff
@@ -361,6 +362,9 @@ Similarly, if you built with ``PARQUET_REQUIRE_ENCRYPTION`` (in C++), you
 need to set the corresponding ``PYARROW_WITH_PARQUET_ENCRYPTION`` environment
 variable to 1.
 
+To set the number of threads used to compile PyArrow's C++/Cython components
+(just like ``-jX``), set the ``PYARROW_PARALLEL`` environment variable.
```

Comment:
I'm not sure the `-jX` reference is generally helpful, if people haven't been using Make before.